### ISSUE TITLE
FrankDoc support for digester rules for self registering elements

### ISSUE THREAD

**Issue by gvanbrakel** [ibissource/iaf#2737]
**❗ please do not add sensitive information in issues, you can provide extra information via email using issue number as reference ❗**

**Describe the issue**
The FrankDoc generator currently cannot handle digester-rules.xml for elements that do not have a direct relation with a parent element, like JmsRealms and SapSystems, and in a way also for Configuration. The FrankDoc generator currently needs a 'registerMethod' attribute to determine from the Java code the type of these elements.
In absence of a 'registerMethod' attribute the FrankDoc generator should determine the type from the 'object' attribute, the 'beanClass' attribute, or from (when necessary new to introduce) 'type' attribute.
The hierarchical location in the configuration file should be determined from the pattern. 
Example configurations:
```
	<rule pattern="configuration/jmsRealms" type="nl.nn.adapterframework.jms.JmsRealms" factory="nl.nn.adapterframework.configuration.digester.JmsRealmsFactory"/>
	<rule pattern="configuration/sapSystems" object="nl.nn.adapterframework.extensions.sap.SapSystems"/>
```

**Reporter**
Gerrit

**Comment by thomaspj10** [comment]
I don't think that this will simplify the current situtation. I agree that adding dummy methods such as `setSapSystems` in `Configuration` is kinda dumb, but automagically determining it would make the relationship between elements even more confusing. There are use cases where these register methods are actually extremly important. A `SenderSeries` for example can contain n senders, while a `SenderWrapper` can only contain one. The set and add methods are used to determine the cardinality. Not including these methods in some use cases and including them in some use cases would make everything even more implicit and confusing.

Another thought I had was adding a `@DigesterRegisterMethod` to all these set and add methods. It doesn't have any functionallity and merely serves as documentation that this method is unused, but is important for the digester. This also has the benefit of easily being able to find these methods using the annotation.
